homeassistant/components/edl21/sensor.py

```diff
@@ -107,16 +131,41 @@ def event(self, message_body) -> None:
                     self._OBIS_BLACKLIST.add(obis)
 
         if new_entities:
-            self._async_add_entities(new_entities, update_before_add=True)
+            self._hass.loop.create_task(self.add_entities(new_entities))
```

Comment:
In Python 3.7 we should use `asyncio.create_task` instead of `loop.create_task`.